### ISSUE TITLE
Intermittent crash after performCollapse

### ISSUE THREAD

**Issue by markrietveld** [splitwise/TokenAutoComplete#125]
Encountered with 1.3.3 (0c42bce62a3cf8c100d80881d9a60de9defe2d3c). I don't know how to reproduce the crash. The crash happens in a [delayed `Runnable`](https://github.com/splitwise/TokenAutoComplete/blob/master/library/src/main/java/com/tokenautocomplete/TokenCompleteTextView.java#L669-669) inside `TokenAutoCompleteTextView`, so I don't know the exact state it was in. The application this runs in does not call `performCollapse` manually
```
Android: 5.1
Manufacturer: asus
Model: Nexus 7

java.lang.IndexOutOfBoundsException: setSpan (17 ... 17) ends beyond length 0
	at android.text.SpannableStringBuilder.checkRange(SpannableStringBuilder.java:1018)
	at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:611)
	at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:607)
	at android.text.Selection.setSelection(Selection.java:76)
	at android.text.Selection.setSelection(Selection.java:87)
	at android.widget.EditText.setSelection(EditText.java:99)
	at com.tokenautocomplete.TokenCompleteTextView$2.run(TokenCompleteTextView.java:662)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5254)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

**Comment by mgod** [comment]
Well, it looks like you're collapsing the view while you have some tokens in it and also editing the content of the field. Is there a place where you are causing the field to lose focus while adding or removing objects?

**Comment by mgod** [comment]
I think I've fixed this in c60d372a45f9197e98a83c7fc5aa7ad1e8d949ce. There was an issue that would allow the user to delete part of the prefix, which would cause crashes whenever we had to do math involving the prefix and the actual text.
